### DISCUSSION TITLE
Data does not update automatically (or only when the app is opened)

### DISCUSSION THREAD

**Discussion by LuxorN09** [niltrip/powerocean#6] [General]
Hello everyone,
First of all, thank you very much for this implementation.
In general it works well, all required values and sensors are set up.
However, I (and others from the German Ecoflow Powerocean Facebook group) have the following problem:
The sensor values are no longer updated irregularly, then it only helps to open the official Ecoflow app on the smartphone once, then the values in Home Assistant are also updated again.
However, this does not only seem to be a problem with the HA integration, the Ecoflow web portal (https://portal.ecoflow.com/) also does not update at this moment, only after starting the smartphone app again.

Is this known or is there anything that can be done?
Thank you!

**Comment by LuxorN09** [comment]
Since last week, the Ecoflow web portal has been updating itself without using the app. 
However, the HA integration only updates the data if the app or the web portal is used in parallel.

**Comment by Bombarde16** [comment]
Hello,
thank you for the integration of Power Ocean.
I have the same problem. The values don´t get any updates.

**Comment by niltrip** [comment]
Still working for me.

![grafik](https://github.com/niltrip/powerocean/assets/34488036/4eec44d6-cf7f-4d0d-921d-afbfb03ae6ec)

**Comment by LuxorN09** [comment]
Addendum: I have been using a PowerInsight for some time now. Since then, the values have been updated automatically. The PI (like the portal and app) is therefore sufficient as a "trigger".

**Reply by andyrozman** [reply]
So you have Raspberry PI running parallel to HA and it tries to connect to web portal (simulating PowerInsight display), which makes values update themselves more regularly?

**Comment by niltrip** [comment]
I have to revise myself. It is actually the case that values are not updated or only in combination with the app. I didn't notice it at the beginning. I probably looked at the app too often ;-). I can't fix that. But I found out that there are currently even more values in the response from the server. Among other things, there are values of the individual strings. It looks like these values are updated more often.

**Comment by hellms** [comment]
moin.
same for me here.
when open die official ecoflow 
app to often during the day, values are updated via this powerocean integration.
if you dont use thier official app, some values are NOT updated.
it seeam it's already a known issue: https://github.com/tolwi/hassio-ecoflow-cloud/discussions/302 

asked ecoflow for joining thier developer program... 
https://developer-eu.ecoflow.com/us
we will see...

cheers

**Reply by hellms** [reply]
quick and dirty workaround fore me is to let run in a tiny linux vm a browser logged in on https://portal.ecoflow.com ...
it runs now for ~two hours and i got updates every few seconds in my HA entities without using the official ecoflow app.

cheers

**Reply by michai-27** [reply]
Here is a solution since September 12th:
https://github.com/tolwi/hassio-ecoflow-cloud/discussions/302

I've got still the problem. Will the solution above integrate in this repository?

**Comment by AndyBowden** [comment]
As reported by @LuxorN09, some users of this integration have reported that some data only updates when one has the EcoFlow smart phone app open.
Using the version of the integration which @niltrip released yesterday, for my Dual Inverter system for which product type 86 applies, I am finding that some entities update when the app is not in use, but some others don’t.
For example, I have 9 entities relating to battery state of charge as listed below.
Of these, 6 are live whether the app is in use or not, but 3 are only live when the app is in use. Those 3 are:
bpSoc_all, bpSoc_master and bpSoc_slave
It would appear that some parts of the JSON response are live at all times, but some are not. I offer this information in the hope that it might assist in identifying which entities can’t be relied on when the app is not in use. I see this as a limitation of the EcoFlow API over which we have no control.
<img width="451" alt="image" src="https://github.com/user-attachments/assets/5895c48e-9b05-45bd-b0a9-c8c939c7bcf4" />

**Comment by niltrip** [comment]
on my branch fix_connection i parallel hold a connection via mqtt.
the idea:
connect via mqtt to Ecoflow as an app. So the values over my way update more often. Can someone confirm if this works?
Thanks.

Before the question "Why did you not using mqtt?". The values over this "private api" are binaries (decode via protobuf or so). To complex for me.

**Reply by hellms** [reply]
moin.
i updated via HACS to pre-release Version 2026.02.12 and copied the content of "fix_connection" on top...
how to proove the mqtt "workaround" ?
cheers

**Reply by niltrip** [reply]
For my understanding, without open the App the values should show changes more often.
Especially sysLoadPwr and sysGridPwr.

**Reply by hellms** [reply]
moin.
in short: nope at all
my actual workaround is a tiny linux "helper" vm with autostart of log into ecoflow web portal.
my HA entities for sysLoadPwr and sysGridPwr gets a refresh every 30secs

with powered off helper vm absolut no update on these entities happen....

i simply overwrote the files in "custom_components/powerocean" with the content of your github "fix_connections" branch.
anything more to do ?

cheers

**Reply by andyrozman** [reply]
@hellms Do you run this tiny linux helper with PI, or do you have VM? Would you share your config? It is something I might need to look into (if this doesn't fix it) when I start using this integration ... (in next few days - Solar Panels are being installed right now, and electrics will be done on thursday)...

**Reply by niltrip** [reply]
@hellms i changed a small thing on the banch and set an other topic. Please try again, or someone else. 
On the other hand, we can implement your doing into the software if it helps.

Furthermore, we might be able to replace the values ​​that aren't updated often with calculated ones. @jdammers  is working on something like that.

As far as I know, the values ​​for the three phases are updated more frequently.. Many data points can be calculated from them...

Btw: Is the change to the device distribution okay?

**Reply by hellms** [reply]
> @hellms Do you run this tiny linux helper with PI, or do you have VM? Would you share your config? It is something I might need to look into (if this doesn't fix it) when I start using this integration ... (in next few days - Solar Panels are being installed right now, and electrics will be done on thursday)...

moin.
nothing high sophisticated, i just used a tiny ubuntu desktop as a vm on proxmox.
autostart/autologin to start browser of your choice and done.
cheers

**Reply by hellms** [reply]
moin niltrip.
short update.
i used the same branch like yesterday. the files are from 2026-02-17 14:38:53
hope these are the right ones... 
again, i just copied them over the existing ones and reload your integration

at the end no change of the behaviour.
no update on sysLoadPwr and sysGridPwr 
updates e.g. on mpptpv_pwrtotal every 60 seconds, which i guess is the default behaviour without any "cheats" ;-)

PS1:
any code for testing around  is welcome ;-)

PS2: 
the seperating of "building blocks" is okay for me.
had the same "evolution" some weeks ago on my daikin onceta integration

**Reply by niltrip** [reply]
Hi, i did not know if the "reload" is the right way. I think an restart would be correct.
Nevertheless, I'm currently running some more tests on my end. I think we're focusing on the wrong values ​​here. Yesterday, I noticed that ecoflow_cloud isn't even offering the values ​​sysloadPwr and sysLoadGrid (I also noticed the same update issue during my experiments with the public API). Last night, I disabled PowerInsight on my system. The values ​​didn't change. However:

pcsMeterPower and pcsActPwr seem to represent the same thing and update more frequently. The same goes for the phases and even the MPPT values.

<img width="2291" height="507" alt="grafik" src="https://github.com/user-attachments/assets/0d38afbf-07a4-479f-b5a1-76026e4a7283" />
<img width="2291" height="523" alt="grafik" src="https://github.com/user-attachments/assets/a2da1e0a-288e-4eae-9d2f-fb96880edf11" />

**Reply by andyrozman** [reply]
@niltrip This might also be dependent on how they wire the system and which (optional) components they use. So theoretically they can use smart meter that is supplied with PO system or they might not. I am just getting my Solar system wired today and they told me that they can wire it different ways, depending on situation... So for example, they will wire CT module near the consumer unit and not where official electricity meter is and that is ok, because consumer unit (fuse box, etc) is 1st stop for my electricity (after official meter). It seems like certain things are optional (additional PO meter for example). So values might not change on your system because of different wiring, but on someone else system they might change.

**Reply by hellms** [reply]
@niltrip 
uh yes, true. there are more entities with values....
i checked and compared:
-  pcsmeterpower vs. sysgridpwr
they are close by. 
pcsmeterpower is updated every 60 seconds. 
for my needs sufficient.

- pcsactpwr vs. sysloadpwr 
they have a real big gap. 
#1 the value of pcsactpwr is negative e.g. -1.850W 🤷‍♀️
#2 if i ignoire the negative, the gap is ~15% between those.
pcsmeterpower is updated very 60 seconds

any ideas about my negative valkues ?

any updates on the to be used branch ?

cheers

**Comment by jdammers** [comment]
@niltrip  @andyrozman, @hellms 
I have been testing the changes I made for a while and would like to share some plots. When I look at the EcoFlow app, the state values in HA are close, although not (always) identical. This may be due to differences in the update time (?).
However, it would be great if someone could test whether the calculations make sense and whether the sensor updates are "reasonable" for you. Although there may be better solutions, please take a look at the code starting at line 661 in ecoflow.py (branch 'fix_stale_sensors').
Below, you will find two snapshots showing only four new sensors that I have added to the 'fix_stale_sensors' testing branch. The sensors update approximately every 60 seconds.

The plots below show the sensor states during periods of low and high solar power production (only four of the new sensors are shown in these examples).

Low solar power consumption:
<img width="1521" height="449" alt="LowSolarPower" src="https://github.com/user-attachments/assets/2e77bbfb-a53d-46d0-8bf3-7ed7b0f931ba" />

High(er) solar power consumption
<img width="1513" height="458" alt="HighSolarPower" src="https://github.com/user-attachments/assets/ea7407c4-63f9-43bf-bcf9-f99bbf0bd64c" />

New sensors
<img width="347" height="512" alt="NewSensors" src="https://github.com/user-attachments/assets/9b275484-9551-43a0-b5bd-f2523fea30ba" />

Note: I am not showing the sensor.powerocean_mpptpv_pwrtotal in these plots.
EDIT: i.e., in the two plots at the top

**Comment by niltrip** [comment]
I ran my test throughout the day today.

My change/idea of ​​logging in via MQTT in parallel isn't working. The login should be correct, but it has no effect on updating the values.

**timeline:**
February 18, 2026 to February 19, 2026, around 12:00 PM: no PowerInsight, no app.
February 19, 2026, from around 12:00 PM: MQTT login in the integration.

**What I've noticed in the process:**
**I mainly see that it affects the values ​​from the data field in the JSON:**
    "data": {
        "sysLoadPwr"
        "sysGridPwr"
        "mpptPwr"
        "bpPwr"
        "bpSoc"
        "sysBatChgUpLimit"
        "sysBatDsgDownLimit"
        "sysGridSta"
        "sysOnOffMachineStat"
        "online"
        "todayElectricityGeneration"
        "monthElectricityGeneration"
        "yearElectricityGeneration"
        "totalElectricityGeneration"
        "systemName"
        "createTime"
        "location"
        "timezone"

**I never used these values.**
        "sysBatChgUpLimit"
        "sysBatDsgDownLimit"
        "sysGridSta"
        "sysOnOffMachineStat"
        "createTime"
        "location"
        "timezone"


**Just an indicator:**
        "online"
        "systemName"

**These are updated less frequently and, in my opinion, don't pose a problem.**
        "todayElectricityGeneration"  (perhaps for this. here i have an configuration.yaml in the repo to integrate over the strings or as calculated parameter. In my opinion a good thing in order to have a change of parameter in the Energy Dashboard (stable interface))
        "monthElectricityGeneration"
        "yearElectricityGeneration"
        "totalElectricityGeneration"

**These are the values ​​we primarily look at, and for them we need an alternative:**
        "sysLoadPwr"
        "sysGridPwr"
        "mpptPwr"
        "bpPwr"
        "bpSoc"

**See also diagrams**
![bpPwr](https://github.com/user-attachments/assets/b330d4c0-3be1-4331-aec9-dfc1b7dc4ea0)
![bpSoc](https://github.com/user-attachments/assets/d758592f-d261-475a-99dd-a274df96c127)
![sysGridPwr](https://github.com/user-attachments/assets/ec58b563-6eb4-4b57-9ac8-2d8bafbb12c3)


"sysLoadPwr" == ???
"sysGridPwr" == pcsMeterPower (or, as I would recommend, centrally at the meter)
"mpptPwr" == mpptPv_pwrTotal (calculated from the individual strings mpptPv1_pwr, mpptPv2_pwr, or mpptPv3_pwr)
"bpPwr" == emsbpPower (battery power (+) to the battery and (-) from the battery) as total power or bpPwr from the individual batteries
"bpSoc" == bpSoc (updates as soon as the percentage value changes) > OK

The parameters
pv1
pv2
pv3
are also updated less frequently.

What I see is that new data is stored in the cloud approximately every 2 hours.
The values ​​for sysLoadPwr, sysGridPwr, mpptPwr, etc., are also available in the ENERGY_STREAM_REPORT, but this makes no difference during the update.

Actually, only one value is missing, and that's the actual household consumption:
"sysLoadPwr" == ???

I'd like to take a closer look at what @jdammers did here and probably implement it.
It should be something like PV power - grid feed-in - battery feed-in.

@hellms 
Your negative value probably represents a feed-in to the grid?
The parameters can be negative or positive.
And thanks again for trying.

@andyrozman 
Yes, I think there can be differences from system to system. For one thing, not everyone might have an off-grid system. I also have the problem that my EcoFlow app shows discrepancies between the consumption readings on my home network and those directly at the meter. This came with an software update...

I hope we now have a good listing of the problematic parameters ;-).

**Reply by niltrip** [reply]
PS.: Something else I've noticed... I can actually turn off PowerInsights, lol.
Now is family time.

**Reply by andyrozman** [reply]
About this replacements... So I have been looking at PO from BLE point of view. sysLoadPwr is updated from machine about every 2-3s (this is what you see on app when you look at house, values change frequently), while for example pcsMeter comes in different message that is not updated so frequently (but I think it still updates at least 1-2 within 10 s window)... Since we call for update from integration every 10s, it might not make such difference from our point of view. But data might not be as exact as that from sysLoadPwr...

One other question... If person has PowerInsight, would then update problem be solved for him/her, since that thing is running all the time (or so I assume).

**Reply by niltrip** [reply]
Yes. I used a powerinsight and did not saw the problem ;-)

**Comment by niltrip** [comment]
@jdammers 
Hi, I'd like to implement your `house_consumption` parameter. We already have the others, right? Or am I missing something? Your parameter perfectly covers the gap when the sun is shining and there's no grid power being drawn.

`pcsMeterPower > grid to house`
`emsbpPower > to/from battery`
`mpptPwr_Total > sun`

I might want to rename it?

`houseLoadPwr`
`housePwr`

or perhaps a slightly abbreviated version of yours.

`Pwr` is especially useful as a trigger for `device_class` and `state_class`.

**Reply by niltrip** [reply]
Okay, from #54 maybe also pcsMeterPower in kWh (grid consumption today > I didn't see it immediately in your code?) and feed-in (probably - pcsMeterPower in kWh ; grid feed-in today) for the dashboard

**Comment by jdammers** [comment]
@niltrip I am fine with your suggestion (including the renaming) if we get the following sensors defined in the main branch.
The definition of the grid import/export sensor was done using a combination of integral and utility meter helpers. I suggest to do this outside of the this integration (see below).
Brief summary of the sensor creation from the fix_sensor_stale branch:
        # ------------------------------
        # From and to Grid sensors
        # From and to Battery sensors
        # ------------------------------
        
        # Base sensors
        solar = float(total_power)
        grid = float(d.get("pcsMeterPower", 0))
        battery = float(d.get("emsBpPower", 0))

        # House Consumption: solar + grid - battery
        house_consumption_raw = solar + grid - battery
        house_consumption = round(max(house_consumption_raw, 0), 1)
        
        # Grid Power (Explizit)
        unique_id = f"{self.sn_inverter}_{report}_grid_power"

        # Grid to Battery: bat_chg - solar_avail_for_bat
        bat_chg = max(battery, 0)
        solar_avail_for_bat = max(solar - house_consumption, 0)
        grid_to_battery = round(max(bat_chg - solar_avail_for_bat, 0), 1)


        # Grid to House: grid import - grid to battery
        grid_import = max(grid, 0)
        grid_to_house = round(max(grid_import - grid_to_battery, 0), 1)


        # Battery to House
        battery_to_house = round(abs(battery) if battery < 0 else 0, 1)


        # Solar to Battery
        solar_to_battery = round(min(solar_avail_for_bat, bat_chg), 1)


        # Solar to Grid
        solar_to_grid = round(max(solar - house_consumption - bat_chg, 0), 1)

        # Solar to House
        solar_to_house = round(min(solar, house_consumption), 1)

     # ============================
     #  Grid import/export sensors
     #  => using helpers
     # ============================
      # Grid import sensor
      1) Integral sensor using sensor.powerocean_grid_power
      2) Utility meter on 1) with daily reset

      # Grid export sensor
      1) Integral sensor using sensor.powerocean_solar_to_grid
      2) Utility meter on 1) with daily reset

**Reply by niltrip** [reply]
@jdammers 
I've added all your sensors to the main branch. However, I would suggest we remove "Grid Power (Explicit)". I understand that you want to keep the other calculated (some explicit) values. Please take another look or try it out. ;-)

**Reply by andyrozman** [reply]
How would house_consumption_raw = solar + grid - battery, give you house consumption... 
On evening I have solar = 0, grid = 0 and battery is filling whole house with its 273 W (or something). So in my case house, per your formula my consumption would be -273 W? 

And be careful pcsMeterPower gives you values for the second that reading was taken, and data refreshes every 10 seconds, so you are missing 9s of data in between... Or at least that would be my assumption on what pcsMeter does?

**Reply by niltrip** [reply]
You're right, there's a missing "absolute" here. I don't see a problem with the second point. At time x, all data points are present and are calculated.

I would agree with you regarding the integration or conversion of values. I also read in HA that the values ​​have to be calculated in the coordinator (which knows when the value is actually available).

By the way:

Could you give me a response from your fetch_data log? If you enable debugging, you get the server response in the logs. You can look at one here: https://github.com/niltrip/powerocean/blob/main/documentation/response_modified.json

In the log file, it's a long line with each fetch. It starts with {"code": "0"..... Just make sure to remove the personal data.

In an editor that can display JSON, you can see it formatted a bit better.

I'd like to check again if Ecoflow has added any new values. I didn't see any solar panels on the last single-phase system...
Thanks.

**Reply by andyrozman** [reply]
@niltrip Was this for me? To get you fetch_data? 

Why would you expect panels on configuration? This is not microinverter system... What you see is Pv1 and Pv2, which would string 1 of panels and string 2 of panels... PO inverter has input for 2 strings and you can't see if one panel is faulty (that would be only possible if you had panels with microinverters, but those wouldn't work with PO)... If one panel is faulty then whole string would be faulty... So for example, my installers split my 12 panels into 2 strings with 6 panels each (I have very small roof)...

**Reply by niltrip** [reply]
@andyrozman 
Yes, was for you ;-). Sorry bad wording.
You don't see any panels, of course. You only see the power output across distributed strings.

I'm just interested in understanding the JSON. I just realized I don't have any data for a single-phase system yet. Otherwise, I'll just be working blindly and saying it works ;-).
I just moved the collection of responses from the documentation to the fixtures section. That way, I can test my code directly against your systems.

My system is a PowerOcean type 83. I don't have PowerGlow or PowerPulse (yet).

**Reply by jdammers** [reply]
@andyrozman @niltrip Long time ago I wrote a script which can export the response output to a json or (friendly reading) yaml file. The script was meant to output the current response but can also be used to compare the current response against a previous response and highlights the differences. 
Recently, I added a REDACT argument. With this argument, it tries to automatically redact private information. 
Let me know if you are interested in this script. I am happy to share it with you. See examples below:

```
"""
This script checks for new parameter sets from EcoFlow PowerOcean API responses and can compare it against previous responses.

Functionality:
- Authenticates with the EcoFlow API and fetches the latest device data.
- Optionally saves the current API response to a file.
- Redacts sensitive data (SNs, location, system name) when saving with `--redact`.
- Compares the current response to a reference JSON file, reporting new, removed, and updated keys/values.
- Groups and formats differences for human-readable and machine-readable output (TXT, YAML, JSON).
- Useful for monitoring changes in the PowerOcean API or device configuration over time.

Usage:
    python3 powerocean_check_response.py [options]

Examples:
    # Save the current API response to a new file with redaction
    python3 powerocean_check_response.py --username 'your@email.com' --password 'yourpassword' --sn 'yoursn'  \
                            --variant 83 --save_new --redact

    # Compare current API response to a reference JSON file
    python3 powerocean_check_response.py --username 'your@email.com' --password 'yourpassword' --sn 'yoursn'  \
                            --variant 83 --fn_json powerocean/Response-EcoFlowAPI_2024-09-25_17-06-15.json

    # Save differences between current and reference response in TXT and JSON
    python3 powerocean_check_response.py --username 'your@email.com' --password 'yourpassword' --sn 'yoursn' \
                            --variant 83 --fn_json powerocean/Response-EcoFlowAPI_2024-09-25_17-06-15.json --save_diff

    # Use YAML format for human-readable diff report
    python3 powerocean_check_response.py --username 'your@email.com' --password 'yourpassword' --sn 'yoursn' \
                             --variant 83 --fn_json powerocean/Response-EcoFlowAPI_2025-08-15.json --save_diff --human_format yaml

For all options, run:
    python3 powerocean_check_response.py --help
"""
```

**Reply by niltrip** [reply]
@jdammers 
Hi, feel free to check it in under documentation. I could have used it myself right now ;-). Did you check the main branch again?

**Reply by jdammers** [reply]
@niltrip I pushed the script under main/documentation. 
No, did not yet check the (changes) in the main branch. I will do it very soon (any specific I should look at?).

**Reply by andyrozman** [reply]
@jdammers So I am total newbie with python, but I can't get this to run... ModuleNotFoundError: No module named 'ecoflow_standalone'. I tried pip install ecoflow_standalone but can't. 

@niltrip I am trying to get file to you, but problem is that I had never had so much problems with any json and I have been working with jsons for about last 8 years (copying from log)... The way it is formatted is very weird... I will see if I can get this directly somehow...

**Reply by andyrozman** [reply]
@niltrip Here is the file, I removed my personal info (I hope)...  
[ef_type28_singlePhase5kw.json](https://github.com/user-attachments/files/25473060/ef_type28_singlePhase5kw.json)

**Reply by niltrip** [reply]
@andyrozman  
Yes, sometimes it's a bit tricky. I like using Notepad++ with the JSON extension. It shows a lot of errors, but it does pretty print.
Reassuring that there are no special features in your JSON.

@jdammers 
Thanks for the script. I'll tweak it a bit so it works with all my changes.
I just wanted you to take another look at your calculations, since I had ​​optimized it. 
Why you want gridPower explicitly again?
the point from [andy](https://github.com/niltrip/powerocean/discussions/6#discussioncomment-15883214) with the -battery in house_consumption. Is there a missing abs or did max it?

**Reply by jdammers** [reply]
@niltrip I apologize if the JSON file format was not compatible with your tools. It was designed to provide a line-by-line output for easy comparison with older API responses. In this way, I can directly follow the output from the terminal (on my MacBook) or in VSCode - while this does not work for me with your JSON file above (I see a single long line only) and with native python I get this error:  JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
The problem may be due to the different operating systems we are using (?)
For the future, if we want to allow users to save their own API responses, we could offer a Powerocean service script that will be called from Developer Tools > Actions > powerocean.save_api_response. This would have the advantage that no credentials would be needed. 

Regarding the **comment from @andyrozman** 
Yes, there is something wrong. I think "battery" should be replaced by "battery_to_house".

Regarding the sensor **"Grid Power (Explizit)"**
The sensor publishes the state of the "grid" variable, which is required for grid import/export calculations, etc. However, we do not really need to publish the sensor state of grid (see block "Grid Power (Explizit)).  
In my version, I published this sensor state, because 'grid' has a signed value (+, -) and I was using it to compare the states of sensors that depend on this variable. However, I agree, we do not really need to published the state of grid although it would not cause any major issues.

**Reply by andyrozman** [reply]
@niltrip Would data structure between "Standard" PO models be different? And by standard I mean, PO that is not Plus or Pro, those two could definitely have a little bit different structure, although looking at BT parsing code, there is almost no difference between Plus and Standard models (only difference I could see was 1 message - EmsChangeReport)

**Reply by andyrozman** [reply]
@niltrip Would you have samples (similar to mine), where HotRod is used, where EV is used and where HeatPump is in use? Like I mentioned I am experimenting with ha-ef-ble project to support powerocean, but at the moment I am missing content for when upper 3 hardware options are used. If I had export I could see what data I need to consider... Doesn't matter for which version of powerocean...

**Reply by niltrip** [reply]
@jdammers 
Yes, I think the JSON isn't true JSON. Notepad++ and the JSON Tools plugin have always worked for me, though.

Thanks for your script. I'll have to make some changes for it. I need to separate the API from Home Assistant again so the script can be called as usual. I'll do that after the release. I monitored your parameters throughout the day again today. That's great. Late last night, I used AI to refine it a bit. Now I can finally finish the release. I'm getting values ​​even though no app or Power Insights are running.

<img width="2282" height="508" alt="Screenshot 2026-02-25 192643" src="https://github.com/user-attachments/assets/914178f7-cc75-4834-aa00-b59a757e3ffd" />

Calling the script as a service is also a good idea. We'll see. Now I want to slowly move the software into my production environment.

@andyrozman 
No, the data isn't significantly different. But since the launch 2.5 years ago, parameters and reports have been added, as well as all the additional devices. I don't have any of them, so I always like to have a few response files. Currently, the repository under tests/fixtures contains various files, including hot rods and EVs. I haven't seen the heat pump yet. I'd be interested in that as well.

**Reply by jdammers** [reply]
@niltrip The plot looks good! I am looking forward to the new release!
In another integration, I implemented such service calls, for example, to also reload the integration without the need for a HA restart, which is very practical.
@andyrozman We will prepare something for easy dumping API responses.

**Reply by andyrozman** [reply]
@niltrip Just one side question... You have been playing around with PO for some time now... Is there any easy way to detect if system is in Backup/Island mode? Is that exposed in integration?

**Reply by niltrip** [reply]
> @niltrip Just one side question... You have been playing around with PO for some time now... Is there any easy way to detect if system is in Backup/Island mode? Is that exposed in integration? 

Not in the integration. I just quickly checked the JSON...

EMS_CHANGE_REPORT
"sysGridSta": 0,

That sounds like a status. But we'd have to see that when there's a failure. I didn't think about that with my last one ;-)

**Comment by CaptainNoBeard** [comment]
I have build something that works for me https://github.com/CaptainNoBeard/ecoflow-keepalive.git

**Reply by hellms** [reply]
cool, tried it immeditaly.
stopped my tiny "helper" vm with browser for ecoflow portal.
looked like had to restart your plugin two times before values are refresh all 60 seconds.
now it works stable for last 12 hours.
letś wait for some days of runtime....

thanks captainnobeard

**Reply by niltrip** [reply]
Is that still necessary? I've now disconnected my PowerInsights.

**Reply by Xygen** [reply]
Only for very few sensors iirc 

I think mpptPwr is affected. But mppt_pwrTotal is a good replacement.

**Reply by andyrozman** [reply]
@niltrip No idea... Haven't installed new version yet... You renamed bunch of sensor which will break all my derived sensors, automations and displays, so I am sort of holding of on doing it... And I use PowerinsightV2, so old sensors seem to work for me still...

@Xygen mpptPwr and mppt_pwrTotal don't show the same values though... Difference is not big between them, so it really depends what you need them for (diff would be around 40W when in 4kW production)...

**Reply by CaptainNoBeard** [reply]
> Is that still necessary? I've now disconnected my PowerInsights.

I do not have PowerInsight. Values do not update for me without watching the app / user portal.

**Reply by hellms** [reply]
> > Is that still necessary? I've now disconnected my PowerInsights.
> 
> I do not have PowerInsight. Values do not update for me without watich the app / user portal.

guess, same here.
what exactly is powerinsight ?
the tablet from ecoflow?
cheers

**Reply by niltrip** [reply]
I actually removed my PowerInsights (yes, the tablet) and I'm getting all the values ​​I need. Please take a look at:

housePower
gridPower
gridToBattery
gridToHouse
batteryToHouse
solarToBattery
solarToGrid
solarToHouse

For the energy dashboard, I created helper parameters from mpptpv1_pwr and mpptpv2_pwr to form an integral (see also configuration.yaml in the repository). With the helper it is matching much better to the batterie energy. For storage, I use bpTotalDsgEnergy and bpTotalChgEnergy. My house meter is separate.

You can also use helper parameters here if you want to use what ecoflow offers.


> @niltrip No idea... Haven't installed new version yet... You renamed bunch of sensor which will break all my derived sensors, automations and displays, so I am sort of holding of on doing it... And I use PowerinsightV2, so old sensors seem to work for me still...
> 
Hopefully not. I've had the same values ​​in my dashboard since 2023. Only the battery values ​​have changed their IDs because they're now assigned to a device. But I haven't used those parameters yet.

<img width="731" height="434" alt="Screenshot 2026-03-18 093911" src="https://github.com/user-attachments/assets/bbfedf2e-0d9d-4183-8abb-9918d85cda08" />

<img width="1686" height="1085" alt="Screenshot 2026-03-18 094550" src="https://github.com/user-attachments/assets/a24f24c4-5dc1-4973-9425-1604a31c6a4e" />

**Reply by CaptainNoBeard** [reply]
Since a few days its brocken, powerocean_mpptpwr and powerocean_sysloadpwr are not updating anymore.
<img width="1352" height="1106" alt="grafik" src="https://github.com/user-attachments/assets/e0ac6822-fb52-4b3b-93a9-64c100d4322d" />

I am currenlty using powerocean_mpptpwr and powerocean_sysloadpwr to calculate solar surplus for ev charging. I have to see what to do.

**Reply by andyrozman** [reply]
Lot of sensors have been renamed... Which is why I am still on old version... I have a lot of derived sensors and they based on old sensors... Nice graph...

**Reply by CaptainNoBeard** [reply]
Hi, 
quick update, it works again. 
<img width="967" height="1089" alt="grafik" src="https://github.com/user-attachments/assets/cfed9b3f-91d2-4c1c-9b23-70b9d990aae3" />
I am not sure what exactly fixed it. I have made some small updates to my [keepalive](https://github.com/CaptainNoBeard/ecoflow-keepalive).
